compiler/packages/babel-plugin-react-compiler/src/Entrypoint/Pipeline.ts

```diff
@@ -272,7 +272,10 @@ function runWithEnvironment(
       validateNoSetStateInRender(hir).unwrap();
     }
 
-    if (env.config.validateNoDerivedComputationsInEffects) {
+    if (
+      env.config.validateNoDerivedComputationsInEffects &&
+      !env.config.validateNoDerivedComputationsInEffects_exp
+    ) {
```

Comment:
maybe merge w the _exp case below? `if (config.validate...exp) { ...} else if (config.validate...) { ... }`